enterprise/server/cmd/ci_runner/main.go

```diff
@@ -808,6 +848,169 @@ func (ar *actionRunner) Run(ctx context.Context, ws *workspace) error {
 	return nil
 }
 
+type runInfo struct {
+	args         []string
+	runfiles     []*bespb.File
+	runfilesRoot string
+}
+
+func uploadRunfiles(ctx context.Context, workspaceRoot, runfilesDir string) ([]*bespb.File, error) {
```

Comment:
we've collectively spent quite a lot of time writing code to upload files to the CAS. Is there a way we could re-use some code we've written before to do this rather than writing this out? Can we use something like UploadDirectoryToCas(/path/to/dir)?